mir-ci/mir_ci/display_server.py

```diff
@@ -33,31 +37,50 @@ def wait_for_wayland_display(runtime_dir: str, name: str) -> None:
             return
     raise RuntimeError('Wayland display ' + name + ' did not appear')
 
-class DisplayServer:
-    def __init__(self, command: Command, add_extensions: Tuple[str, ...] = ()) -> None:
-        self.command = command
+class DisplayServer(Benchmarkable):
+    def __init__(self, app: Dependency, add_extensions: Tuple[str, ...] = ()) -> None:
+        self.command: Command = cast(Command, app.command)
+        self.command_type = app.app_type
         self.add_extensions = add_extensions
         # Snaps require the display to be in the form "waland-<number>". The 00 prefix lets us
         # easily identify displays created by this test suit and remove them in bulk if a bunch
         # don't get cleaned up properly.
         self.display_name = 'wayland-00' + str(os.getpid())
 
-    def program(self, command: Command, env: Dict[str, str] = {}) -> Program:
+    async def get_cgroup(self) -> Cgroup:
+        return await self.server.get_cgroup()
+    
+    @staticmethod
+    def needs_systemd_run(app_type: Optional[DependencyType]) -> bool:
+        if app_type == "deb" or app_type == "pip":
+            return True
+
+        return False
+
+    def program(
+            self, 
+            command: Command,
+            env: Dict[str, str] = {},
+            app_type: Optional[DependencyType] = None) -> Program:
         return Program(command, env=dict({
                 'DISPLAY': 'no',
                 'QT_QPA_PLATFORM': 'wayland',
                 'WAYLAND_DISPLAY': self.display_name
             },
-            **env)
-        )
+            **env),
+            with_systemd_run=DisplayServer.needs_systemd_run(app_type))
```

Comment:
I don't think DisplayServer should have anything specific here… just pass `app_type` down to `Program` and let it DTRT?